app/services/user_opportunity_discovery.py

```
@@ -364,6 +378,111 @@ def _compute_strategy_fingerprint(self, strategies: List[Dict[str, Any]]) -> str
+
+        for candidate in candidates:
+            if candidate in owned_ids:
+                return candidate
+        return candidates[0] if candidates else None
+
+    def _looks_like_uuid(self, value: str) -> bool:
+        """Return True when value appears to be a UUID."""
+
+        try:
+            uuid.UUID(str(value))
+            return True
```

Ruff (0.14.0)

Consider moving this statement to an `else` block

(TRY300)